docs/source/tutorials/datasets.rst

```diff
@@ -0,0 +1,457 @@
+              ]
+            )
+
+Creating a dataset API key with specific rights
+-----------------------------------------------
+
+Via the |sdk| you can create a dataset-specific API key.
+The API key is one way to :ref:`authentication:Authenticate with Encord`.
+You need to provide the ``dataset_hash``, which uniquely identifies a dataset (see, e.g., the :ref:`tutorials/datasets:Listing existing datasets` to get such hash).
+If you haven't created a dataset already, you can have a look at :ref:`tutorials/datasets:Creating a Dataset`.
+
+.. autolink-concat:: section
+
+.. autolink-preface::
+    from encord.orm.dataset_api_keys import DatasetAPIKey
+    from encord.orm.dataset import DatasetScope
+
+
+.. tabs::
+
+    .. tab:: Code
+
+        .. literalinclude:: /code_examples/tutorials/datasets/creating_a_dataset_api_key.py
+            :language: python
+
+    .. tab:: Example output
+
+        .. code-block:: python
+
+            DatasetAPIKey(
+              dataset_hash="<dataset_hash>",
+              api_key="<api_key>",
+              title="Example api key title",
+              scopes=[
+                  DatasetScope.READ,
+                  DatasetScope.WRITE,
+              ]
+            )
+
+
+.. note::
+
+    This capability is available to only the Admin of a dataset.
+
+
+With the API key at hand, you can use :ref:`authentication:API key authentication`.
+
+
+Fetching dataset API keys
+-------------------------
+
+Via the |sdk|, you can get all API keys for an existing dataset.
+You need to provide the ``dataset_hash`` which uniquely identifies a dataset (see, e.g., the :ref:`tutorials/datasets:Listing existing datasets` to get such hash).
+If you haven't created a dataset already, you can have a look at :ref:`tutorials/datasets:Creating a Dataset`.
+
+
+.. autolink-concat:: section
+
+.. autolink-preface::
+    from encord.orm.dataset_api_keys import DatasetAPIKey
+    from encord.orm.dataset import DatasetScope
+
+
+.. tabs::
+
+    .. tab:: Code
+
+        .. literalinclude:: /code_examples/tutorials/datasets/fetching_dataset_api_keys.py
+            :language: python
+
+    .. tab:: Example output
+
+        .. code-block:: python
+
+            [
+                DatasetAPIKey(
+                  dataset_hash="<dataset_hash>",
+                  api_key="<dataset_api_key>",
+                  title="Full Access API Key",
+                  scopes=[
+                      DatasetScope.READ,
+                      DatasetScope.WRITE,
+                  ]
+                ),
+                # ...
+            ]
+
+With the API key at hand, you can use :ref:`authentication:API key authentication`.
+
+
+Data
+====
+
+Adding data
+-----------
+
+You can add data to datasets in multiple ways.
+You can both use |company| storage, as described next, and you can :ref:`add data from a private cloud <tutorials/datasets:Adding data from a private cloud>` to retain complete control of the data.
```

Comment:
```suggestion
You can both use |company| storage, as described next, or you can :ref:`add data from a private cloud <tutorials/datasets:Adding data from a private cloud>` to integrate any pre-existing data.
```

Otherwise it heavily implies one is better than the other!